emsdk/patches/0002-library-eventloop-signatures.patch

```diff
@@ -0,0 +1,150 @@
+     '$runtimeKeepalivePush', '$runtimeKeepalivePop',
+ #endif
+   ],
++  emscripten_set_immediate__sig: 'iii',
+   emscripten_set_immediate: function(cb, userData) {
+     polyfillSetImmediate();
+     {{{ runtimeKeepalivePush(); }}}
+@@ -89,6 +90,7 @@ LibraryJSEventLoop = {
+     '$runtimeKeepalivePop',
+ #endif
+   ],
++  emscripten_clear_immediate__sig: 'vi',
+   emscripten_clear_immediate: function(id) {
+     {{{ runtimeKeepalivePop(); }}}
+     emClearImmediate(id);
+@@ -99,6 +101,7 @@ LibraryJSEventLoop = {
+     '$runtimeKeepalivePush', '$runtimeKeepalivePop',
+ #endif
+   ],
++  emscripten_set_immediate_loop__sig: 'vii' ,
+   emscripten_set_immediate_loop: function(cb, userData) {
+     polyfillSetImmediate();
+     function tick() {
+@@ -119,6 +122,7 @@ LibraryJSEventLoop = {
+     '$runtimeKeepalivePush', '$runtimeKeepalivePop',
+ #endif
+   ],
++  emscripten_set_timeout__sig: 'iidi',
+   emscripten_set_timeout: function(cb, msecs, userData) {
+     {{{ runtimeKeepalivePush() }}}
+     return setTimeout(function() {
+@@ -129,6 +133,7 @@ LibraryJSEventLoop = {
+     }, msecs);
+   },
+ 
++  emscripten_clear_timeout__sig: 'vi',
+   emscripten_clear_timeout: function(id) {
+     clearTimeout(id);
+   },
+@@ -138,6 +143,7 @@ LibraryJSEventLoop = {
+     '$runtimeKeepalivePush', '$runtimeKeepalivePop',
+ #endif
+   ],
++  emscripten_set_timeout_loop__sig: 'vidi',
+   emscripten_set_timeout_loop: function(cb, msecs, userData) {
+     function tick() {
+       var t = performance.now();
+@@ -162,6 +168,7 @@ LibraryJSEventLoop = {
+     '$runtimeKeepalivePush', '$runtimeKeepalivePop',
+ #endif
+   ],
++  emscripten_set_interval__sig: 'iidi',
+   emscripten_set_interval: function(cb, msecs, userData) {
+     {{{ runtimeKeepalivePush() }}}
+     return setInterval(function() {
+@@ -174,10 +181,11 @@ LibraryJSEventLoop = {
+ #if !MINIMAL_RUNTIME
+   emscripten_clear_interval__deps: ['$runtimeKeepalivePop'],
+ #endif
++  emscripten_clear_interval__sig: 'vi',
+   emscripten_clear_interval: function(id) {
+     {{{ runtimeKeepalivePop() }}}
+     clearInterval(id);
+   },
+ };
+ 
+-mergeInto(LibraryManager.library, LibraryJSEventLoop);
++mergeInto(LibraryManager.library, LibraryJSEventLoop);
+\ No newline at end of file
+
+From aa4e361c57266b1b18c3035a73cc2075fadf2745 Mon Sep 17 00:00:00 2001
+From: John Wason <wason@wasontech.com>
+Date: Sat, 18 Jun 2022 20:37:35 -0400
+Subject: [PATCH 2/2] Use "p" for pointers in signatures, and restore newline
+ at end of file
+
+---
+ src/library_eventloop.js | 12 ++++++------
+ 1 file changed, 6 insertions(+), 6 deletions(-)
+
+diff --git a/src/library_eventloop.js b/src/library_eventloop.js
+index 2f232900dcd..5cf835d9973 100644
+--- a/src/library_eventloop.js
++++ b/src/library_eventloop.js
+@@ -73,7 +73,7 @@ LibraryJSEventLoop = {
+     '$runtimeKeepalivePush', '$runtimeKeepalivePop',
+ #endif
+   ],
+-  emscripten_set_immediate__sig: 'iii',
++  emscripten_set_immediate__sig: 'ipp',
```

Comment:
These changes replacing 'i' with 'p' surely don't matter for us: they would be a problem for the wasm64 target but we don't use it and it doesn't work yet.

Reply:
Ah I see it was requested upstream. That's fine then.